skglm/tests/test_intercept.py

```diff
@@ -0,0 +1,111 @@
+from skglm.solvers.cd_solver import cd_solver, cd_solver_path
```

Comment:
Ideally I would avoid to duplicate this test file with the one from `test_estimator.py`

Reply:
+1. Why do we need an extra `test_intercept.py`? Can't we parametrize all tests in `test_estimator.py` with `fit_intercept`?

Reply:
agreed, this needs to be addressed